### DISCUSSION TITLE
Priced in BTC or sats

### DISCUSSION THREAD

**Discussion by HeadyWook** [btcpayserver/btcpayserver#2250] [Ideas / Feature Requests]
I am thinking of a way for a user to set the price of a product in USD, say $25, and have its value displayed in BTC or sats rather than USD. Perhaps the price updates every 3-5 mins, upon page refresh, etc.?

Currently, users can set a USD value for a product which displays in BTC/satoshis once the "pay" button is hit and the QR code is generated. Users can also set a BTC/satoshis value for a product (i.e., change the denomination from USD to BTC/sats) but it remains static and does not update with price movement. For example, at current rates, $25 of a $36,056.17 bitcoin is 0.00069336 BTC, or 69,336 sats. If a user set the value of a product at 69,336 sats, that sat value would remain the same despite the value of bitcoin denominated in USD. The price of the product in sats will not update to price movements. For example, if the price of bitcoin reaches $40,000 (the same problem holds true if price drops too), the set value of 69,336 sats will no longer be equivalent in value as $25. Now $25 of a $40,000 bitcoin would be 62,500 sats, a difference of 6,836 sats. 

This discussion is about a method for a user to set a USD value in the back-end but display in real-time a BTC or sat equivalent to the customer. No matter the price of bitcoin in USD, the BTC or sat value displayed will adjust to equate the USD value determined by the user.

Please comments, questions, or concerns.

**Comment by dennisreimann** [comment]
I also thought about this and it would be a nice feature to have. I imagine a websocket connection that updates the prices displayed whenever the rate provider updates.

**Reply by dennisreimann** [reply]
Now that I talked to @kukks about it and he informed me that the rate provider isn't a background service (as I assumed), I think the best way to achieve this is using the API and having a time interval in which to fetch and update the prices via that.

**Comment by Kukks** [comment]
There is already a rates API. The pay button should and needs to be simplistic in nature.  It is highly customizable due to it being a simple html form element. You can construct the html form elements manually in real time while you update the rate through the API.

**Reply by HeadyWook** [reply]
> 
> 
> There is already a rates API. The pay button should and needs to be simplistic in nature. It is highly customizable due to it being a simple html form element. You can construct the html form elements manually in real time while you update the rate through the API.

I am not following. Can you explain it again?

Here is what I mean. See image below. The price label is "$" and the price is set to "35". If I change the price label to "sats" and the price to "88,785" (which is currently equivalent to $35), then if the bitcoin price changes, it will still show 88,785 sats. What I am discussing is a feature for the price to be denominated in sats or BTC and it adjusts for changes in price. Today it might display 8,785 sats and tomorrow it could be 5,878, but it would still be equivalent to $35.

![image](https://user-images.githubusercontent.com/38931215/107104678-93c93a00-67d7-11eb-887d-7accbdd2ac05.png)
